enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/FileUserRealm.java

```diff
@@ -270,6 +283,63 @@ boolean deleteUser( String username ) throws IOException
         return result;
     }
 
+    void suspendUser( String username ) throws IOException
+    {
+        synchronized ( this )
+        {
+            User user = userRepository.findByName( username );
+            if ( user == null )
+            {
+                throw new IllegalArgumentException( "User " + username + " does not exist." );
+            }
+            if ( !user.hasFlag( IS_SUSPENDED ) )
+            {
+                User suspendedUser = user.augment().withFlag( IS_SUSPENDED ).build();
+                try
+                {
+                    userRepository.update( user, suspendedUser );
+                }
+                catch ( ConcurrentModificationException e )
+                {
+                    // Due to the synchronized (this) we should never get this exception unless the user repository
+                    // is modified outside this neo instance. In that case we do not know how to proceed.
+                    throw new RuntimeException( "Unexpected failure", e );
+                }
+            }
+        }
+    }
+
+    void activateUser( String username ) throws IOException
+    {
+        synchronized ( this )
+        {
+            User user = userRepository.findByName( username );
+            if ( user == null )
+            {
+                throw new IllegalArgumentException( "User " + username + " does not exist." );
+            }
+            if ( user.hasFlag( IS_SUSPENDED ) )
+            {
+                User activatedUser = user.augment().withoutFlag( IS_SUSPENDED ).build();
+                try
+                {
+                    userRepository.update( user, activatedUser );
+                }
+                catch ( ConcurrentModificationException e )
+                {
+                    // Due to the synchronized (this) we should never get this exception unless the user repository
+                    // is modified outside this neo instance. In that case we do not know how to proceed.
+                    throw new RuntimeException( "Unexpected failure", e );
```

Comment:
I think we should follow the same pattern as with all the other methods and just retry (*unconditionally) and not introduce unnecessary assumptions about how this repository is being used at this point. (*The likelihood of an endless race between multiple entities modifying the same repository record concurrently should be reasonably slim)